### PR TITLE
Add document about caching in matomo.

### PR DESCRIPTION
**Pull request by diosmosis**

### CODE REVIEW

File: docs/4.x/caching-in-matomo.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+list in multiple places in PHP, but only really need to issue a database query once.
+
+See the {@see Matomo\Cache\Backend\Transient::class} class for more information.
+
+## Lazy Cache
+
+The **Lazy** cache is a cache that requests and reads cached data on demand. When you cache data in the Lazy cache,
+it will be available across requests, for example, in a file backend, but will not be read until it is requested.
+
+In the backend, the data will be stored in one entry per cache key (eg, one php file per cache key).
```
Comment (Findus23) [inline docs/4.x/caching-in-matomo.md]:
```suggestion
In the backend, the data will be stored in one entry per cache key (e.g., one php file per cache key).
```

File: docs/4.x/caching-in-matomo.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+cache to be accessed only when we want the data. The Lazy cache is perfect for this.
+
+See the {@see Matomo\Cache\Backend\Lazy::class} class for more information.
+
+## Eager Cache
+
+The **Eager** cache is a cache like the Lazy cache in that it will make cached data available across requests, but
+every piece of cached data in the Eager cache is loaded at the start of each request.
+
+In the backend, the eager cache data will all be stored in a single entry (so, eg, all cached data is in a single php
```
Comment (Findus23) [inline docs/4.x/caching-in-matomo.md]:
```suggestion
In the backend, the eager cache data will all be stored in a single entry (so, e.g., all cached data is in a single php
```

File: docs/4.x/caching-in-matomo.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+Each individual cache type has a different way of doing that.
+
+# Cache Backends
+
+The cache types above all store data in the same place. This place is determined by the cache backend specified in INI configuration.
+The `[Cache] backend` option determines where the data gets saved. The possible options are straightforward: cache to an **array**,
+cache to a **file**, cache to **redis**.
+
+There is also the **chained** cache which is worth a note. The chained cache allows using multiple backends. For example,
+caching to an array, file and redis. Here, when requesting data, we'd first check if it's in the array backend, and if there use it.
```
Comment (Findus23) [inline docs/4.x/caching-in-matomo.md]:
```suggestion
caching to an array, file and redis. Here, when requesting data, we'd first check if it's in the array backend, and if it is there, we use it.
```

**Review by Findus23 — CHANGES_REQUESTED** [review]
Just some minor style things

**Comment by diosmosis** [comment]
@Findus23 :+1: made those changes

**Comment by tsteur** [comment]
Great 👍  thanks @diosmosis at some point could think about making it a public API

**Pull request merged**